### PR TITLE
Add NUnit 3 Test Adapter

### PR DESCRIPTION
**Pull request by DasSkelett**
Add NUnit 3 Test Adapter for running NUnit tests in VisualStudio.

### CODE REVIEW

File: Tests/Tests.csproj [hunk 1]
```diff
@@ -203,4 +206,10 @@
     <Exec Command="powershell ../build.ps1 Generate-GlobalAssemblyVersionInfo" Condition="!Exists('../_build/meta/GlobalAssemblyVersionInfo.cs') And '$(OS)' == 'Windows_NT'" />
     <Exec Command="sh ../build Generate-GlobalAssemblyVersionInfo" Condition="!Exists('../_build/meta/GlobalAssemblyVersionInfo.cs') And '$(OS)' == 'Unix'" />
   </Target>
+  <Target Name="EnsureNuGetPackageBuildImports" BeforeTargets="PrepareForBuild">
+    <PropertyGroup>
+      <ErrorText>Dieses Projekt verweist auf mindestens ein NuGet-Paket, das auf diesem Computer fehlt. Verwenden Sie die Wiederherstellung von NuGet-Paketen, um die fehlenden Dateien herunterzuladen. Weitere Informationen finden Sie unter "http://go.microsoft.com/fwlink/?LinkID=322105". Die fehlende Datei ist "{0}".</ErrorText>
```
Comment (HebaruSan) [inline Tests/Tests.csproj]: Any chance you could get the English version of this error instead?

Comment (DasSkelett) [reply Tests/Tests.csproj]: Oh I see...
I'll try!

File: Tests/Tests.csproj [hunk 2]
```diff
@@ -12,6 +13,8 @@
     <FileAlignment>512</FileAlignment>
     <OutputPath>..\_build\out\$(AssemblyName)\$(Configuration)\bin\</OutputPath>
     <IntermediateOutputPath>..\_build\out\$(AssemblyName)\$(Configuration)\obj\</IntermediateOutputPath>
+    <NuGetPackageImportStamp>
+    </NuGetPackageImportStamp>
```
Comment (HebaruSan) [inline Tests/Tests.csproj]: Are these lines needed? They don't look like they would do much since there's no content between them.

Comment (DasSkelett) [reply Tests/Tests.csproj]: Yeah, don't know, Visual Studio inserted them itself.
I'll try to delete them.

**Comment by HebaruSan** [comment]
(My 30 day trial of VS Community apparently expired, and I'm reinstalling it to try this out...)

**Comment by DasSkelett** [comment]
I thought VS Community ist fully free?
At least I'm paying nothing and don't reinstall monthly :D

**Comment by HebaruSan** [comment]
It seems that it's both free, _and_ has a 30 day trial period (and reinstalling doesn't work):

image

I guess they want me to create a Microsoft account. At that point I'm willing to trust that it works on your system :grin: .

**Comment by DasSkelett** [comment]
Oh that's so typical :smile: 
Well, it worked even after a restart of VS, so I think we can assume it working every time.
All tests passed successfully (dunnow about those `[EXPLICIT]` ones, didn't try them).

**Pull request merged**